### PR TITLE
fix: preserve GitHub transport contracts after adapter deepening

### PR DESCRIPTION
**Pull request by moomooskycow**
## Reviewer Evidence
- Start here: `Manual QA` and `Walkthrough` below.
- Direct video download: N/A (internal behavior fix; no UI surface).
- Walkthrough notes: `master` already moved GitHub review-path reads and writes behind `github_platform`; this branch is now the small follow-up that restores caller-visible behavior on top of that deeper boundary.
- Fast claim: This PR keeps the deeper adapter from `#364` while restoring three contracts: entry-accurate `stop_on_marker`, `fetch_comments(...)=[]` on invalid payloads, and `lib.github` exception types from `github_reviews`.

## Why This Matters
- Problem: After `master` absorbed the deeper GitHub platform adapter, three caller-visible behaviors still drifted: `stop_on_marker` stopped at the page rather than the matching entry, `fetch_comments` no longer normalized invalid payloads to `[]`, and `github_reviews` leaked raw platform exceptions to callers like `post-verdict-review.py`.
- Value: The deeper boundary stays intact, but existing callers keep the stable behavior they already depend on.
- Why now: Rebasing onto `master` narrowed this PR from an architectural refactor to a contract-preservation follow-up, so the right merge target is now the behavioral gaps left on top of `#364`.
- Issue: N/A (follow-up to the GitHub platform boundary refactor).

## Trade-offs / Risks
- Value gained: Stable caller contracts, precise stop-marker semantics, and lower regression risk for review posting flows.
- Cost / risk incurred: `github.py` and `github_reviews.py` each keep a thin exception-remapping wrapper on top of the deeper platform boundary.
- Why this is still the right trade: The wrappers are the public boundary callers already know; pushing raw platform exceptions outward would leak implementation detail and break existing error handling.
- Reviewer watch-outs: Remaining risk is mostly live GitHub API behavior under real permission/rate-limit conditions, not local control flow.

## What Changed
This PR no longer competes with the platform-boundary refactor on `master`. It is now the compatibility layer that keeps that deeper boundary while restoring the public `lib.github` behavior existing scripts expect.

### Base Branch
```mermaid
graph TD
  A["lib.github.fetch_comments"] --> B["github_platform.fetch_issue_comments"]
  A --> C["ValueError escapes on invalid payload"]
  D["lib.github_reviews"] --> E["github_platform review helpers"]
  E --> F["Raw platform exceptions"]
  G["stop_on_marker"] --> H["Returns full page through marker"]
```

### This PR
```mermaid
graph TD
  A["lib.github.fetch_comments"] --> B["github_platform.fetch_issue_comments"]
  A --> C["ValueError maps to []"]
  D["lib.github_reviews"] --> E["github_platform review helpers"]
  D --> F["Platform exceptions remapped to lib.github exceptions"]
  G["stop_on_marker"] --> H["Returns at matching entry"]
```

### Architecture / State Change
```mermaid
stateDiagram-v2
  [*] --> DeepAdapter
  DeepAdapter --> ContractDrift: "Caller behavior changed"
  ContractDrift --> StableAdapter: "Restore public wrapper contract"
```

Why this is better:
- The deeper module boundary from `master` remains untouched.
- Marker cut-off behavior now matches the documented stopping point.
- `post-verdict-review.py` and similar callers keep catching the exception types they already expect.
- Invalid GitHub comment payloads degrade back to `[]` instead of surfacing an unexpected decode failure.

<details>
<summary>Intent Reference</summary>

## Intent Reference
- Keep the deeper GitHub platform boundary from ADR 004.
- Preserve stable caller behavior while the implementation boundary moves underneath it.

</details>

<details>
<summary>Changes</summary>

## Changes
- `scripts/lib/github_platform.py`
- Make `stop_on_marker` return immediately after appending the matching entry instead of returning the rest of the page.
- `scripts/lib/github.py`
- Restore the legacy empty-list fallback when the shared transport returns an invalid JSON payload.
- `scripts/lib/github_reviews.py`
- Remap platform permission/transient errors back to `lib.github.CommentPermissionError` and `lib.github.TransientGitHubError`.
- `tests/test_github.py`
- Add regression coverage for permission translation, transient translation, and invalid-payload fallback.
- `tests/test_github_platform.py`
- Tighten marker tests so entries after the marker on the same page are excluded.
- `tests/test_github_reviews.py`
- Add regression coverage for review-helper exception translation.

</details>

<details>
<summary>Acceptance Criteria</summary>

## Acceptance Criteria
- [x] The deeper GitHub platform adapter on `master` remains intact.
- [x] `stop_on_marker` stops at the matching entry, not the page.
- [x] `lib.github.fetch_comments` returns `[]` on invalid payloads.
- [x] `github_reviews` preserves the `lib.github` exception contract for callers.
- [x] Regressions are covered by focused tests.

</details>

<details>
<summary>Alternatives Considered</summary>

### Option A: Do nothing
- Upside: No more churn on top of `#364`.
- Downside: Keep caller-visible regressions in review posting and marker handling.
- Why rejected: The deeper boundary is only worth keeping if existing scripts still behave correctly on top of it.

### Option B: Expose raw platform exceptions everywhere
- Upside: Fewer wrapper lines.
- Downside: Forces every caller to learn the platform boundary and breaks current error handling.
- Why rejected: Weakens information hiding and pushes migration cost onto unrelated callers.

### Option C: Keep the deep boundary and restore public wrapper contracts (chosen)
- Upside: Best compatibility-to-diff ratio.
- Downside: Keeps thin remapping logic in wrapper modules.
- Why chosen: Small, explicit, and preserves the module boundary already merged to `master`.

</details>

<details>
<summary>Manual QA</summary>

## Manual QA
- Command: `uvx --from pytest pytest tests/test_github.py tests/test_github_platform.py tests/test_github_reviews.py tests/test_post_verdict_review.py tests/test_collect_overrides.py -q`
- Result: `103 passed in 0.17s`
- Command: `uvx --from pytest pytest tests/test_github.py tests/test_github_platform.py tests/test_github_reviews.py -q`
- Result: `60 passed in 0.07s`
- Command: `uvx ruff check scripts/lib/github.py scripts/lib/github_platform.py scripts/lib/github_reviews.py tests/test_github.py tests/test_github_platform.py tests/test_github_reviews.py tests/test_post_verdict_review.py`
- Result: `All checks passed!`
- Command: `find scripts tests -name '*.sh' -type f -exec shellcheck {} +`
- Result: no findings

</details>

<details>
<summary>Walkthrough</summary>

## Walkthrough
- Renderer: Terminal walkthrough
- Artifact: Command transcript in `Manual QA` plus the before/after diagrams above.
- Claim: The branch preserves the deeper adapter on `master` while restoring wrapper-level behavior that review posting flows still depend on.
- Before / After scope: `lib.github`, `lib.github_platform`, `lib.github_reviews`, and the related focused tests.
- Persistent verification: `uvx --from pytest pytest tests/test_github.py tests/test_github_platform.py tests/test_github_reviews.py -q`
- Residual gap: Full end-to-end GitHub API behavior still depends on CI and live token permissions.
- Observable artifact from branch execution:
```text
$ uvx --from pytest pytest tests/test_github.py tests/test_github_platform.py tests/test_github_reviews.py -q
60 passed in 0.07s

$ uvx ruff check scripts/lib/github.py scripts/lib/github_platform.py scripts/lib/github_reviews.py tests/test_github.py tests/test_github_platform.py tests/test_github_reviews.py tests/test_post_verdict_review.py
All checks passed!
```

</details>

<details>
<summary>Before / After</summary>

## Before / After
- Before: after rebasing onto `#364`, this branch conflicted with a base that had already merged the deeper adapter, and the open PR still described the old refactor instead of the remaining compatibility gaps.
- After: the branch is reduced to a small follow-up that restores entry-accurate marker behavior, invalid-payload fallback, and review-helper exception translation on top of the already-merged adapter.

</details>

<details>
<summary>Test Coverage</summary>

## Test Coverage
- Updated: `tests/test_github.py`
- Updated: `tests/test_github_platform.py`
- Updated: `tests/test_github_reviews.py`
- Regression-adjacent verification: `tests/test_collect_overrides.py`, `tests/test_post_verdict_review.py`
- Preserved by rebase: the standalone `scripts/lib/github.py --help` regression in `tests/test_github.py` still passes.

</details>

<details>
<summary>Merge Confidence</summary>

## Merge Confidence
- Confidence level: High
- Strongest evidence: GitHub `Quality Checks` passed on March 12, 2026, `103` focused local tests passed across the affected GitHub/verdict surface, `60` focused transport/comment/review tests passed, `ruff` passed, `shellcheck` passed, and local `diff-cover` for the patch is now `100%`.
- Remaining uncertainty: Live GitHub API behavior under production permission/rate-limit edge cases, plus local `make validate` reproduction still depends on matching the repo's Python test environment.
- What could still go wrong after merge: A live GitHub API response shape or permission edge not covered by focused tests could still differ from local assumptions, or contributors may need the repo's expected Python test setup to reproduce the same full local gate behavior.

</details>

### CODE REVIEW

File: scripts/lib/github.py [hunk 1]
```diff
@@ -75,40 +69,14 @@ def fetch_comments(
...
-
-        if stop_on_marker is not None:
-            for comment in payload:
-                if isinstance(comment, dict) and stop_on_marker in str(comment.get("body", "")):
-                    return comments
-
-        if len(payload) < per_page:
-            break
-    return comments
+    """Fetch all issue comments for a PR (paginated)."""
```
[the remark this answers is not in the document]
Comment (phrazzld) [reply scripts/lib/github.py]: Restored in `e9e405b`.

`fetch_comments()` now has its detailed public docstring again, including argument and return details plus the updated marker semantics. I kept that explicit wrapper contract because callers still depend on this module-level API even though the transport implementation moved underneath it.

File: scripts/lib/github_reviews.py [hunk 2]
```diff
@@ -10,7 +10,7 @@
 import tempfile
 from dataclasses import dataclass
 
-from lib import github as gh
+from lib.github_platform import run_gh
```
[the remark this answers is not in the document]
Comment (phrazzld) [reply scripts/lib/github_reviews.py]: Fixed in `e9e405b`.

I handled both pieces of drift here:
- `github_reviews.py` now wraps `platform_run_gh()` with a local adapter that re-raises the `lib.github` exception types expected by `post-verdict-review.py`.
- `fetch_issue_comments()` now stops at the matching entry instead of returning later comments from the same page.

Added regression tests for the review-side exception remap, the `fetch_comments()` wrapper contract, and the same-page marker cutoff.

**Comment by phrazzld** [comment]
Addressed in `e9e405b`.

The follow-up fix keeps the shared pagination boundary but corrects the two review findings from the previous commit:
- `stop_on_marker` now returns immediately after the matching comment is appended, so entries later on the same page are excluded.
- `fetch_comments()` and `github_reviews.py` now preserve the `lib.github` exception contract expected by existing callers.

Focused regression coverage was added for both behaviors.
